internal/newusers/cache/db.go

```diff
@@ -221,34 +200,31 @@ func (c *Cache) Close() error {
 	return c.db.Close()
 }
 
+// Clear removes the database file.
+func Clear(cacheDir string) error {
+	return os.Remove(filepath.Join(cacheDir, dbName))
+}
+
 // ClearAndRebuild closes the db and reopens it.
-func (c *Cache) ClearAndRebuild(cacheDir string) {
+func (c *Cache) ClearAndRebuild(cacheDir string) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
 	if err := c.db.Close(); err != nil {
-		slog.Warn(fmt.Sprintf("Could not close database %v", err))
+		return fmt.Errorf("could not close database: %v", err)
+	}
+
+	if err := os.Remove(filepath.Join(cacheDir, dbName)); err != nil && !errors.Is(err, fs.ErrNotExist) {
```

Comment:
And here, I would call `c.Remove()`

Reply:
you did +1 but this one was not changed, was it?

Reply:
Oh, I reverted the diff because I had removed the error check as well and forgot to reupdate the os.Remove call.